src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -54,87 +63,75 @@ const defaultProps = {
     bankAccountList: [],
     cardList: [],
     isLoadingPayments: false,
+    enableSelection: false,
```

Comment:
Ah nevermind I see there is another. 

But sorry, I think this is possibly not the correct component to use here - look at other "selectable lists" we have. It is better to create a `SelectablePaymentMethodList` than add complexity to the relatively simple `PaymentMethodList`.

Reply:
If I will use OptionList. then I will to have more code such as sections creation, Move the same method of preparing the payment methods, etc to that as well.

Instead, I thought to refactor the existing getPaymentMethods in the paymentMethodList component which behaves the same as before but usages a consistent data structure `PaymentMethod` for all types of payment methods. It makes it easier to pass any payment method (bank, card) to any function that accepts a paymentMethod. 
Mostly, I wanted to use ID instead of `fundID` and `backAccountID`

The only added behaviour is
1. Show checkBoxes.
2. Add filtering and based on that change the title of `Add payment Method` button title.

I refactored it as passing `NativeEvent` first param which is not used was confusing.  Now It takes two callbacks one for when existing payment Method is clicked and second which is called when add payment method is clicked.

Reply:
I think maybe the assignment was not understood here. `SelectablePaymentMethodList` should wrap `PaymentMethodList` not duplicate the contents.